### DISCUSSION TITLE
Revising card names and descriptions for en & de

### DISCUSSION THREAD

**Discussion by SebastianHira** [sumpfork/dominiontabs#550] [General]
Hey, I'd like to revise the whole jsons for cards. Given I only can speak english and german, I'd stick with those. There are a few more questions before I begin.

1. For current and future ease, I'd re-organize the whole file by expansion->alphabet instead of global alphabet. It really is a pain to look up all cards the current way. Especially since the integrity ofd the old system was broken by another user who started a new alphabet at the bottom. Also, using https://wiki.dominionstrategy.com/index.php/List_of_cards enables users to quickly update the file, at least for english. For other languages, it is easier to work down the file with the website.
 
2. I don't think I have the expertise to edit all extras. Descriptions are easily copied. I could copy the extras from the rulebooks, but those could be outdated by erratas. I think either way, that they have no priority.

3. How to deal with the renaming of Harem? Name it "Harem / Farm"?

Also, this has nothing to do with the cards description, but cards_bd has Tide Pools wrongly priced (3 instead of 4). I'd upload this file aswell, so just pulling it would be needed.

**Comment by sumpfork** [comment]
JSON keys are automatically sorted by the pre commit hooks, and I'd like to keep it that way exactly so that I don't have to check each contribution for not messing up some otherwise unenforced sorting.

**Reply by SebastianHira** [reply]
I see. I think this wouldnt be too much of an issue, since it the website allows cards to be sorted alphabetically aswell.
But the files seem to be very messy in general. There are multiple things I don't understand why they are the way they are, such as many keys not being capitalized. Just seems to be no notation agreed upon.

I think I'll just stick to updating what is missing in german first, like boons, prophecies, guilds & Cornucopia 2nd ed, so prople can at least have working german translated tabs. But before theres any rewriting there should really be some notation structure

**Reply by SebastianHira** [reply]
Could you help me with how to push new commits/ create pull requests? I am completely new to github, installing repos locally etc. I haven't really worked out how to install the script even, I think - I've managed to download the repo, but using any commands doesn't work. Didn't use "pip install domdiv" because I didnt know where to enter that to begin with.
A dumbed down quick install might be nice to have in the installation tab - Given you need zero code experience to edit some text in a language file. 

For the commit, the files are ready. This is what i've done:
Updated types_en_us.json to include Coffers
Updated types_de.json to have all keys used in types_en_us.json
Updated Tide Pools price in cards_db.json
Several Notation fixes in both cards_de.json and cards_en_us.json
Added missing cards to cards_de.json
Started updating base game for cards_de.json to new errata but then I got lazy (Someone could check all cards just to be sure though, at least every old expansion up to Adventures would really profit from an update of their errata/new versions)

**Reply by nickv2002** [reply]
@SebastianHira GitHub is a hugely popular platform used by millions of developers so there are tons of great resources on how to use it across the web (much better than we could provide for you here). Here's [GitHub's official "Hello World" getting started page](https://docs.github.com/en/get-started/start-your-journey/hello-world), but I encourage you to do a simple web search to ask "how do I do _some thing_ on GitLab" and you'll find lots of guides and videos etc.

The Python tooling used for this project is admittedly pretty advanced even for a Python user, but the neat thing it will run tests etc for you automatically if you create a Pull Request and give you some feedback on what might be wrong. 

If you have a passing familiarity with Docker you can also use the Docker image to abstract away the Python install bits ([see readme](https://github.com/sumpfork/dominiontabs?tab=readme-ov-file#docker)). I might recommend that rather than managing your own version of Python also. If you're new to Docker it's also very popular so there's lots of resources for it: here's the [official intro](https://docs.docker.com/get-started/introduction/).

**Comment by sumpfork** [comment]
What Nick said, but also let me be clear:

I'm afraid it's just not true that you need zero technical experience to change the language files. They are in a specific format that can break the script if you don't follow it. You should be at minimum be able to package the files, run the linters and tests to make a working pull request. Anything else just creates a huge amount of work on my side to fix up people's submissions - I don't have time for that, sadly. 

We can ask for help with submission other than working PRs, but that'll rely on help from the community unless they require only small fixes.

**Comment by mbbush** [comment]
I've started work on an automated script that pulls down the latest card text from the wiki and rewrites the json files in their expected format. I have plenty of technical experience with python, github, etc, but the only human language I speak fluently is English, with limited French and Spanish but no German. Once I get English working, I could probably extend it to other languages, but I'd have no good way to validate the result myself.
